flatfs.go

```diff
@@ -38,37 +40,52 @@ var _ datastore.Datastore = (*Datastore)(nil)
 
 type ShardFunc func(string) string
 
-func New(path string, getDir ShardFunc, sync bool) (*Datastore, error) {
-	fs := &Datastore{
-		path:   path,
-		getDir: getDir,
-		sync:   sync,
+func New(path string, fun0 string, sync bool) (*Datastore, error) {
```

Comment:
I didn't do this yet.  Do you basically want:
```
func New(path string, sync bool) (*Datastore, error) {
  NewWithFunc(path, "auto", sync)
}

func NewWithFunc(path string, fun0 string, sync bool) {...}
```

Is it that you don't like the special "auto" value?  I don't want to eliminate the use of that string as it makes writing the conversion function/utility easier, so using two separate functions won't really simplify any code.